core/executor/src/host_interface.rs

```diff
@@ -788,9 +789,11 @@ impl_wasm_host_interface! {
 				_ => return Ok(3),
 			};
 
-			context.write_memory(pubkey_data, &pubkey.serialize()[1..65])
-				.map_err(|_| "Invalid attempt to set pubkey in ext_secp256k1_ecdsa_recover")?;
-
+			match compressed {
+				0 => context.write_memory(pubkey_data, &pubkey.serialize()[1..65]),
```

Comment:
why not `if compressed == 0 {} else {}`?

Then we don't have this new return value `4`